src/arith/rewrite_simplify.h

```diff
@@ -221,6 +221,8 @@ class RewriteSimplifier::Impl : public IRMutatorWithAnalyzer {
   bool CanProveGreaterEqual(const PrimExpr& x, int64_t val) {
     return analyzer_->CanProveGreaterEqual(x, val);
   }
+  // Whether the lanes are scalable
+  bool ScalableLanes(const PrimExpr& lanes) { return !lanes.as<IntImmNode>(); }
```

Comment:
Seconding the stricter checks here.  As it is, this function would state that `Mul(2, 3)` is a scalable expression, because it is something other than an integer.

Reply:
Agreed, I'll change it to explicitly check for `vscale`.